File: twidere/src/main/java/org/mariotaku/twidere/view/PreferencesItemIconView.java
package org.mariotaku.twidere.view;

import android.content.Context;
import android.support.annotation.NonNull;
import android.support.annotation.Nullable;
import android.util.AttributeSet;

import org.mariotaku.chameleon.Chameleon;
import org.mariotaku.chameleon.ChameleonUtils;

/**
 * Created by mariotaku on 2016/12/23.
 */

public class PreferencesItemIconView extends IconActionView {
    public PreferencesItemIconView(Context context) {
        super(context);
    }

    public PreferencesItemIconView(Context context, AttributeSet attrs) {
        super(context, attrs);
    }

    public PreferencesItemIconView(Context context, AttributeSet attrs, int defStyleAttr) {
        super(context, attrs, defStyleAttr);
    }

    @Nullable
    @Override
    public Appearance createAppearance(@NonNull Context context, @NonNull AttributeSet attributeSet, @NonNull Chameleon.Theme theme) {
        Appearance appearance = new Appearance();
        appearance.setActivatedColor(ChameleonUtils.getColorDependent(theme.getColorControlActivated()));
        appearance.setDefaultColor(theme.getColorForeground());
        return appearance;
    }
}

File: twidere/src/main/kotlin/org/mariotaku/twidere/view/PreferencesItemIconView.kt
package org.mariotaku.twidere.view

import android.content.Context
import android.util.AttributeSet

import org.mariotaku.chameleon.Chameleon
import org.mariotaku.chameleon.ChameleonUtils
import org.mariotaku.twidere.view.iface.IIconActionButton

/**
 * Created by mariotaku on 2016/12/23.
 */
class PreferencesItemIconView(context: Context, attrs: AttributeSet? = null) : IconActionView(context, attrs) {

    override fun createAppearance(context: Context, attributeSet: AttributeSet, theme: Chameleon.Theme): IIconActionButton.Appearance? {
        val appearance = IIconActionButton.Appearance()
        appearance.activatedColor = ChameleonUtils.getColorDependent(theme.colorControlActivated)
        appearance.defaultColor = theme.colorForeground
        return appearance
    }
}
